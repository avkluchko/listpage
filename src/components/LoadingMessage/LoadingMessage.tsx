import React from 'react';

type LoadingMessageProps = {} & typeof defaultProps;

const defaultProps = {
    text: 'Загрузка данных...'
};

const LoadingMessage = ({ text }: LoadingMessageProps) => {
    return (
        <p>{text}</p>
    );
};

LoadingMessage.defaultProps = defaultProps;

export default LoadingMessage;
