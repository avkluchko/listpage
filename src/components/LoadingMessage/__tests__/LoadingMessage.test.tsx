import React from 'react';
import { render } from '@testing-library/react';

import LoadingMessage from '../LoadingMessage';

describe('LoadingMessage', () => {
    it('should take a snapshot', () => {
        const { asFragment } = render(<LoadingMessage/>);
        expect(asFragment()).toMatchSnapshot();
    });

    it('should renders correctly with default text', () => {
        const { getByText, getByRole } = render(<LoadingMessage/>);
        getByText('Загрузка данных...');

        const indicator = getByRole('progressbar');
        expect(indicator).toHaveStyle('width: 22px; height: 22px');
    });

    it('should renders correctly with custom text', () => {
        const { getByText } = render(<LoadingMessage text="Custom Text"/>);
        getByText('Custom Text');
    });

    it('should renders correctly with custom size', () => {
        const { getByRole } = render(<LoadingMessage size={10}/>);

        const indicator = getByRole('progressbar');
        expect(indicator).toHaveStyle('width: 10px; height: 10px');
    });
})
