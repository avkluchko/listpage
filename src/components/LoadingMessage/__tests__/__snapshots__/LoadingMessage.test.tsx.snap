// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingMessage should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="makeStyles-main-1"
  >
    <div
      class="MuiCircularProgress-root makeStyles-indicator-2 MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
      role="progressbar"
      style="width: 22px; height: 22px;"
    >
      <svg
        class="MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
          cx="44"
          cy="44"
          fill="none"
          r="20.2"
          stroke-width="3.6"
        />
      </svg>
    </div>
    <p
      class="MuiTypography-root MuiTypography-body2"
    >
      Загрузка данных...
    </p>
  </div>
</DocumentFragment>
`;
